apps/console/src/features/identity-providers/components/settings/connected-apps.tsx

```diff
@@ -0,0 +1,529 @@
+    const [ connectedAppsCount, setconnectedAppsCount ] = useState<number>(0);
+    const [ isAppsLoading, setIsAppsLoading ] = useState<boolean>(true);
+    const [ searchQuery, setSearchQuery ] = useState<string>("");
+    const featureConfig: FeatureConfigInterface = useSelector((state: AppState) => state.config.ui.features);
+    const applicationTemplates: ApplicationTemplateListItemInterface[] = useSelector(
+        (state: AppState) => state.application.templates);
+    const groupedApplicationTemplates: ApplicationTemplateListItemInterface[] = useSelector(
+        (state: AppState) => state?.application?.groupedTemplates);
+    const [
+        isApplicationTemplateRequestLoading,
+        setApplicationTemplateRequestLoadingStatus
+    ] = useState<boolean>(false);
+
+    const { t } = useTranslation();
+
+    useEffect(() => {
+        setIsAppsLoading(true);
+        getIDPConnectedApps(editingIDP.id)
+            .then(async (response: ConnectedAppsInterface) => {
+                setconnectedAppsCount(response.count);
+                
+                if (response.count > 0) {
+                    
+                    const appRequests: Promise<any>[] = response.connectedApps.map((app: ConnectedAppInterface) => {
+                        return getApplicationDetails(app.appId);
+                    });
+
+                    const results: ApplicationBasicInterface[] = await Promise.all(
+                        appRequests.map(response => response.catch(error => {
+                            dispatch(addAlert({
+                                description: error?.description
+                                || "Error occurred while trying to retrieve connected applications.",
+                                level: AlertLevels.ERROR,
+                                message: error?.message || "Error Occurred."
+                            }));
+                        }))
+                    );
+
+                    setConnectedApps(results);
+                    setFilterSelectedApps(results);
+                }
+            })
+            .catch((error) => {
+                dispatch(addAlert({
+                    description: error?.description
+                    || "Error occurred while trying to retrieve connected applications.",
+                    level: AlertLevels.ERROR,
+                    message: error?.message || "Error Occurred."
+                }));
+            })
+            .finally(() => {
+                setIsAppsLoading(false);
+            });
+    }, []);
+
+    /**
+     * Fetch the application templates if list is not available in redux.
+     */
+    useEffect(() => {
+        if (applicationTemplates !== undefined) {
+            return;
+        }
+    
+        setApplicationTemplateRequestLoadingStatus(true);
+    
+        ApplicationTemplateManagementUtils.getApplicationTemplates()
+            .finally(() => {
```

Comment:
Let's add `catch` to capture the errors as well

Reply:
Added in https://github.com/wso2/identity-apps/pull/3494/commits/ce8c570994579ea30e3ab52031945bf056094375